v2/internal/testcommon/vcr/redact.go

```diff
@@ -16,6 +16,22 @@ import (
 	"github.com/Azure/azure-service-operator/v2/internal/testcommon/creds"
 )
 
+type Redactor struct {
+	azureIDs   creds.AzureIDs
+	redactions map[string]string
```

Comment:
Suggest renaming this to `literalRedactions` - better indicates what it does, and leaves the door open for adding `regexRedactions` later (which I suspect we'll need).

Reply:
Actually, I've got rid of the map finally, and using a struct + supporting regexp replacements too. Took suggestions from the pointers you shared.